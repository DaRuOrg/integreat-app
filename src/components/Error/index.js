import React from 'react'
import PropTypes from 'prop-types'
import { translate } from 'react-i18next'
import FontAwesome from 'react-fontawesome'

import { history } from 'main'

import style from './style.css'

class Error extends React.Component {
  static propTypes = {
    error: PropTypes.string.isRequired
  }

  /**
   * Go back in history!!
   * @param event The click event
   */
  goBack = (event) => {
    event.preventDefault()
    history.goBack()
  }

  render () {
    const {t} = this.props
    return <div>
      <div className={style.centerText}>{ t(this.props.error) }</div>
      <div className={style.centerText}><FontAwesome name='frown-o' size="5x"/></div>
      <a className={style.centerText} href="/" onClick={this.goBack}>Go back</a>
    </div>
  }
}

export default translate('errors')(Error)
