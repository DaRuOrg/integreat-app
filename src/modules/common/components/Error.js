import React from 'react'
import PropTypes from 'prop-types'
import { translate } from 'react-i18next'
import FontAwesome from 'react-fontawesome'

import style from './Error.css'

export class Error extends React.Component {
  static propTypes = {
    error: PropTypes.string.isRequired
  }

  render () {
    return <div>
      <div className={style.centerText}>{this.props.t(this.props.error)}</div>
      <div className={style.centerText}><FontAwesome name='frown-o' size='5x' /></div>
      <a className={style.centerText} href='/' onClick={this.goBack}>Go back</a>
    </div>
  }
}

export default translate('errors')(Error)
