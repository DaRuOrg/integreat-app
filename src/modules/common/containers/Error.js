import React from 'react'
import PropTypes from 'prop-types'
import { translate } from 'react-i18next'
import FontAwesome from 'react-fontawesome'
import { goBack } from 'redux-little-router'
import { connect } from 'react-redux'
import compose from 'lodash/fp/compose'

import style from './Error.css'

export class Error extends React.Component {
  static propTypes = {
    error: PropTypes.string.isRequired,
    goBack: PropTypes.func.isRequired
  }

  /**
   * Go back in history!!
   * @param event The click event
   */
  goBack (event) {
    event.preventDefault()
    this.props.goBack()
  }

  render () {
    return <div>
      <div className={style.centerText}>{this.props.t(this.props.error)}</div>
      <div className={style.centerText}><FontAwesome name='frown-o' size='5x' /></div>
      <a className={style.centerText} href='/' onClick={this.goBack}>Go back</a>
    </div>
  }
}

const mapDispatchToProps = (dispatch) => ({
  goBack: () => dispatch(goBack())
})

export default compose(
  connect(() => {}, mapDispatchToProps),
  translate('errors')
)(Error)
