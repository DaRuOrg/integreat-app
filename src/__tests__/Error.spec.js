import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import Error from '../components/Error'

jest.mock('store')
jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate HoC receive the t function as a prop
  translate: () => Component => props => <Component t={() => ''} {...props} />
}))

describe('Error', () => {
  test('should match should', () => {
    const component = renderer.create(
      <Error error="Error Message"/>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  test('call history when onBack is called', () => {
    const preventDefault = jest.fn()
    const history = require('store').history

    const wrapper = shallow(<Error error="Error Message"/>).dive()

    wrapper.instance().goBack({preventDefault})

    expect(history.goBack.mock.calls).toHaveLength(1)
    expect(preventDefault.mock.calls).toHaveLength(1)
  })
})
