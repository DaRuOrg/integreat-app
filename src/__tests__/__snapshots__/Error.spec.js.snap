// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should match should 1`] = `
<div>
  <div
    className="centerText"
  >
    
  </div>
  <div
    className="centerText"
  >
    <span
      aria-hidden={true}
      className="fa fa-frown-o fa-5x"
    />
  </div>
  <a
    className="centerText"
    href="/"
    onClick={[Function]}
  >
    Go back
  </a>
</div>
`;

exports[`Error should match snapshot 1`] = `
<div>
  <div
    className="centerText"
  >
    
  </div>
  <div
    className="centerText"
  >
    <span
      aria-hidden={true}
      className="fa fa-frown-o fa-5x"
    />
  </div>
  <a
    className="centerText"
    href="/"
    onClick={[Function]}
  >
    Go back
  </a>
</div>
`;
